metrics/metrics-tests/src/test/java/org/apache/james/metrics/tests/RecordingMetricFactoryTest.java

```diff
@@ -0,0 +1,102 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.metrics.tests;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.metrics.api.Metric;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.jayway.awaitility.Duration;
+import com.jayway.awaitility.core.ConditionFactory;
+
+class RecordingMetricFactoryTest {
+
+    private static final String TIME_METRIC_NAME = "timerMetric";
+    private static final String METRIC_NAME = "metric";
+    private static final java.time.Duration ONE_SECOND = java.time.Duration.ofSeconds(1);
+    private static final java.time.Duration FIVE_SECONDS = java.time.Duration.ofSeconds(5);
+    private static final ConditionFactory WAIT_CONDITION = await().timeout(Duration.ONE_MINUTE);
+
+    private RecordingMetricFactory testee;
+
+    @BeforeEach
+    void setUp() {
+        testee = new RecordingMetricFactory();
+    }
+
+    @Test
+    void executionTimesForATimeMetricShouldBeStoreMultipleTime() {
+        runTimeMetric(ONE_SECOND);
+        runTimeMetric(FIVE_SECONDS);
+
+        WAIT_CONDITION
```

Comment:
This wait is clearly not needed as we can write this test in a synchronous monothread fashion:

```
TimeMetric timer1 = metricFactory.timer(NAME);
Thead.sleep(ONE_SECOND);
timer1.stopAndPublish();

TimeMetric timer2 = metricFactory.timer(NAME);
Thead.sleep(FIVE_SECOND);
timer2.stopAndPublish();

assertThat(testee.executionTimesFor(TIME_METRIC_NAME)).hasSize(2);
assertThat(testee.executionTimesFor(TIME_METRIC_NAME)).at(0).isGreaterThat(ONE_SECOND);
assertThat(testee.executionTimesFor(TIME_METRIC_NAME)).at(1).isGreaterThat(FIVE_SECOND);
```